src/psyclone/transformations.py

```diff
@@ -2430,29 +2394,27 @@ def __str__(self):
 
     @property
     def name(self):
-        '''Returns the name of this transformation as a string.'''
+        '''
+        :returns the name of this transformation.
+        :rtype: str
+        '''
```

Comment:
Just a quick comment from me: in PSyData I'am using:
```
return self.__class__.__name__
```
in the base class, and got rid of ``name`` in all derived classes (and similarly for ``__str__``). Since you are introducing a new baseclass, I thought that might be useful (it would probably change some names, eg. ``GOLoopSwap`` would become ``GOLoopSwapTrans``). Though maybe it could be used even in ``Transform`` instead of ``LoopTrans``/